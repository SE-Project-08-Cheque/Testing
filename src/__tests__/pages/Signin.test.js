// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import Signin from '../../pages/signin';

afterEach(cleanup);



it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Signin />, div)
})

it("renders sign in correctly", () => {
    const {getByText} = render(<Signin />)
    // fireEvent.click(getByText("Sign In"))
    expect(getByText("Sign In")).toBeInTheDocument();
    expect(getByText("Sign in")).toBeInTheDocument();
    expect(getByText("Sign in")).toHaveTextContent("Sign in");
})
it("click button correctly", () => {
    const {getByText} = render(<Signin />)
    // const {queryByTestId} = render(<Signin />)
    fireEvent.click(getByText("Sign In"))
    // fireEvent.change(queryByTestId("username_input"),{ target: { value: 'chalindu' } })
    // fireEvent.click(getByTestId("signin-button"))
})

it("check whether inputs works correctly", () => {
    const {getByTestId} = render(<Signin />)
    const fieldUsername = getByTestId('username_input').querySelector('input');
    fireEvent.change(fieldUsername,{ target: { value: 'chalindu' } })
    expect(fieldUsername.value).toBe("chalindu");
    
    const fieldPassword = getByTestId('password_input').querySelector('input');
    fireEvent.change(fieldPassword,{ target: { value: 'password' } })
    expect(fieldPassword.value).toBe("password");
})

// describe("changeInInput", () => {
//         it("onChange", () => {
//             const {queryByTitle} = render(<Signin />)
//             const input = queryByTitle("Username");
//             console.log(input)
//             fireEvent.change(input, {target: { value: "testValue"}}); //change input to testValue. Like we gave an input
//             expect(input.value).toBe("testValue");
//         })
//     })

it("matches snapshot", () => {
    const tree = renderer.create(<Signin />).toJSON();
    expect(tree).toMatchSnapshot();
})