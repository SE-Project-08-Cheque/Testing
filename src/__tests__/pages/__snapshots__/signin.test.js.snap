// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
      data-testid="signin-text"
    >
      Sign in
    </h1>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      data-testid="username_input"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="Username"
        id="Username-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="Username"
          name="Username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Username
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      data-testid="password_input"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="password"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Password
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p>
        
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
      data-testid="signin-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Sign In
      </span>
    </button>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="#"
          onBlur={[Function]}
          onFocus={[Function]}
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="./signup"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Don't have an account? Sign Up
        </a>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-13"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      2021, All Rights Reserved | Designed by University of Moratuwa
    </p>
  </div>
</main>
`;
